### PR TITLE
Run kubectl as a docker container

### PR DESCRIPTION
**Pull request by asymmetric**
Towards  https://github.com/giantswarm/k8scloudconfig/issues/136

### CODE REVIEW

File: templates.go [hunk 1]
```diff
@@ -740,20 +740,20 @@ write_files:
...
+      while [ "$(/usr/bin/docker run --net=host --rm $IMAGE get cs | grep Healthy | wc -l)" -ne "3" ]; do sleep 1 && echo 'Waiting for healthy k8s'; done
 
       # apply calico CNI
       CALICO_FILES="calico-configmap.yaml calico-node-sa.yaml calico-policy-controller-sa.yaml calico-ds.yaml calico-policy-controller.yaml"
 
       for manifest in $CALICO_FILES
       do
           while
-              /opt/bin/kubectl --kubeconfig=/etc/kubernetes/config/kubelet-kubeconfig.yml apply -f /srv/$manifest
+              /usr/bin/docker run --net=host --rm $IMAGE apply -f /srv/$manifest
```
Comment (calvix) [inline templates.go]: you need to mount `/srv` folder in order to work

File: templates.go [hunk 2]
```diff
@@ -780,7 +780,7 @@ write_files:
       for manifest in $MANIFESTS
       do
           while
-              /opt/bin/kubectl --kubeconfig=/etc/kubernetes/config/kubelet-kubeconfig.yml apply -f /srv/$manifest
+              /usr/bin/docker run --net=host --rm $IMAGE apply -f /srv/$manifest
```
Comment (calvix) [inline templates.go]: you need to mount `/srv` folder in order to work

**Comment by xh3b4sd** [comment]
Just to make sure, this has to be updated in the operators before https://github.com/giantswarm/kubernetesd/pull/164 is merged/deployed.

**Comment by asymmetric** [comment]
I think either way, there will be a small window where clusters won't work, but I'll follow your suggested order.

**Comment by xh3b4sd** [comment]
You are probably right. We just broke the contract. Never mind.

**Pull request merged**